pkg/storage/gcp/bucket_cloud.go

```diff
@@ -58,5 +58,5 @@ func (b *gcpBucket) List(ctx context.Context, prefix string) ([]string, error) {
 
 func (b *gcpBucket) Exists(ctx context.Context, path string) bool {
 	_, err := b.Object(path).Attrs(ctx)
-	return err == nil
+	return err == storage.ErrObjectNotExist
```

Comment:
Should this be `!=`?

If the error was nil, the result will be false. This will return true only when the object does not exist.

Reply:
@robjloranger if err nil, the result *should* be false. And if err is netiher nil nor notExist, then we'll leave it up to the getter to find out what the error is when they try to get that object. Is that ok? Otherwise, we have to make the `Exists` in `Checker` interface return an `error`. What do you think?

Reply:
Sounds good. @robjloranger let me know if this pr looks ok then will merge.

Reply:
@robjloranger you're right i should be doing a more thorough check: 
```golang
	_, err := b.Object(path).Attrs(ctx)
	if err == nil {
          return true
        } else if err == storage.ErrObjectNotExist {
           return false
        } 
        
        TODO: this means that something errored out and we should return false, err
        return false
```

Reply:
what do you think?

Reply:
Awesome 😀

Reply:
@robjloranger ok i've updated the PR :) let me know if it's good to merge! 
PS. we should start having a GCP set up in our CI, what do you think ?